community/codegen/src/main/java/org/neo4j/codegen/ExpressionTemplate.java

```diff
@@ -219,16 +222,17 @@ protected void templateAccept( CodeBlock method, ExpressionVisitor visitor )
         return expressions;
     }
 
-    static ExpressionTemplate invokeSuperConstructor( final ExpressionTemplate[] parameters )
+    //TODO I am not crazy about the way type parameters are sent here
```

Comment:
Would you care to elaborate?

1. What is it you don't like about this way?
2. Do you have any idea on how you would like it to be done?

Could it possibly be done using only a single `parameters` array, if the `ExpressionTemplate` has a type? (if a different type than the actual evaluated type of the expression is used in the signature of the constructor, we could have some sort of `expr.asType( typeRef )` method)